crates/gem_evm/src/uniswap/path.rs

```diff
@@ -0,0 +1,150 @@
+use std::collections::HashSet;
+
+use super::FeeTier;
+use crate::address::EthereumAddress;
+
+use alloy_core::primitives::Bytes;
+use alloy_primitives::aliases::U24;
+use primitives::{AssetId, EVMChain};
+
+pub struct TokenPair {
+    pub token_in: EthereumAddress,
+    pub token_out: EthereumAddress,
+    pub fee_tier: FeeTier,
+}
+
+impl TokenPair {
+    pub fn new_two_hop(token_in: &EthereumAddress, intermediary: &EthereumAddress, token_out: &EthereumAddress, fee_tier: &FeeTier) -> Vec<TokenPair> {
+        vec![
+            TokenPair {
+                token_in: token_in.clone(),
+                token_out: intermediary.clone(),
+                fee_tier: fee_tier.clone(),
+            },
+            TokenPair {
+                token_in: intermediary.clone(),
+                token_out: token_out.clone(),
+                fee_tier: fee_tier.clone(),
+            },
+        ]
+    }
+}
+
+pub struct BasePair {
+    pub native: EthereumAddress,
+    pub stable: EthereumAddress,
+}
+
+impl BasePair {
+    pub fn to_set(&self) -> HashSet<EthereumAddress> {
+        HashSet::from([self.native.clone(), self.stable.clone()])
+    }
+}
+
+impl From<AssetId> for EthereumAddress {
+    fn from(asset_id: AssetId) -> Self {
+        let token_id = asset_id.token_id.unwrap_or_default();
+        EthereumAddress::parse(&token_id).unwrap()
+    }
+}
+
+pub fn get_base_pair(chain: &EVMChain) -> Option<BasePair> {
```

Comment:
let's support multiple stable coin values, for eth also WBTC

Reply:
added, will modify the request part later